library/StandardLibrary/src/Juvix/Library/Feedback.hs

```diff
@@ -0,0 +1,101 @@
+module Juvix.Library.Feedback
+  ( Feedback (..),
+    FeedbackT (..),
+    feedback,
+    feedbackT,
+    Fail.fail,
+  )
+where
+
+import qualified Control.Monad.Fail as Fail
+import qualified Control.Monad.Trans as Trans
+import qualified Data.String as String
+import Juvix.Library
+
+-- | Keep track of messages during the compilation or REPL.
+-- The implementation is based on
+-- [[https://github.com/UU-ComputerScience/uu-cco/blob/master/uu-cco/src/CCO/Feedback.hs]].
+data Feedback (app :: * -> *) msg a
+  = Success (app msg) a -- Indicate success.
+  | Fail (app msg) -- Indicate a failure.
```

Comment:
I don't think it's a good idea having the same `msg` for both Success and Fail.

It'd be great to have a datatype that encapsulates all pipeline errors. E.g.
```
data PipelineError = ParseError ... | TypecheckError ... | CompileError ... | ...
```

What do you think?

Reply:
Mind that the current message implemented as a string is something to be improved with new implementations of the functions in the pipeline.